packages/docusaurus-theme-common/src/utils/codeBlockUtils.ts

```diff
@@ -23,18 +23,23 @@ const commentPatterns = {
 
 type CommentType = keyof typeof commentPatterns;
 
-const magicCommentDirectives = [
-  'highlight-next-line',
-  'highlight-start',
-  'highlight-end',
-];
+export type MagicCommentConfig = {
+  className: string;
+  line?: string;
+  block?: [string, string];
```

Comment:
What about?

```suggestion
  block?: {start: string, end: string};
```

Both works but I think using named attributes makes it easier to read/understand downstream

Reply:
I was trying to mimic TextMate grammar, but they seem to be inconsistent as well, sometimes using tuples and sometimes with objects...